src/common/inject/inject.jsx

```diff
@@ -524,8 +517,42 @@ Zotero.Inject = new function() {
 				title: title
 			}
 		);
+
+		if (options.snapshot) {
+			try {
+				let { pageData, form } = await Zotero.SingleFile.retrievePageData();
+				data["pageData"] = pageData;
+				form["payload"] = JSON.stringify(data);
+				data = form;
+			}
+			catch (e) {
+				Zotero.debug("Fail retrieving page data");
+				Zotero.Messaging.sendMessage("progressWindow.done", [false, 'unexpectedError']);
+				throw e;
```

Comment:
Actually, I guess I take back my previous comment — SingleFile failing shouldn't cause the whole save to fail. It'd be better to fall back to saving the attachment in the client, if we can do so easily.

Reply:
Ok. That's still the the one piece I haven't upgrade to SingleFileZ. I was going to work on that today.